src/main/java/com/twingly/search/Query.java

```diff
@@ -1,94 +1,228 @@
 package com.twingly.search;
 
+import java.text.SimpleDateFormat;
 import java.util.Date;
+import java.util.Objects;
+
+import static com.twingly.search.Constants.SEARCH_QUERY_DATE_FORMAT;
 
 /**
  * The type Query.
  */
 public class Query {
+    private static final SimpleDateFormat QUERY_DATE_FORMAT = new SimpleDateFormat(SEARCH_QUERY_DATE_FORMAT);
     private Date startTime;
     private Date endTime;
-    private String searchPattern;
-    private String documentLanguage;
+    private String searchQuery;
+    private String lang;
+    private String location;
 
     /**
      * Package-level constructor for API usage only. API users should use QueryBuilder instead.
      *
      * @see QueryBuilder
+     * @deprecated since 3.0.0 - use {@link #Query(Date, Date, String, String, String)} instead
      */
+    @Deprecated
     Query() {
         //
     }
 
+    Query(Date startTime, Date endTime, String searchQuery, String lang, String location) {
+        this.startTime = startTime;
+        this.endTime = endTime;
+        this.searchQuery = searchQuery;
+        this.lang = lang;
+        this.location = location;
+    }
+
+    /**
+     * Gets language
+     *
+     * @return language
+     * @since 3.0.0
+     */
+    public String getLang() {
+        return lang;
+    }
+
+    /**
+     * Sets language
+     *
+     * @param lang language
+     * @since 3.0.0
+     */
+    public void setLang(String lang) {
+        this.lang = lang;
+    }
+
+    /**
+     * Convert current query object to #QueryBuilder.
+     *
+     * @return QueryBuilder object from current Query
+     * @since 3.0.0
+     */
+    public QueryBuilder toBuilder() {
+        return QueryBuilder.fromQuery(this);
+    }
+
+    /**
+     * Gets location
+     *
+     * @return the location
+     * @since 3.0.0
+     */
+    public String getLocation() {
```

Comment:
Remove `get/setLocation` (they aren't used anywhere). Location is part of the query string.